src/dataflow/src/sink/tail.rs

```diff
@@ -42,18 +42,14 @@ pub fn tail<G>(
                     while cursor.val_valid(&batch) {
                         let row = cursor.val(&batch);
                         cursor.map_times(&batch, |time, diff| {
-                            assert!(*diff >= 0, "negative multiplicities sinked in tail");
-                            let diff = *diff as usize;
+                            assert!(*diff == 1, "non-unary diff sinked in tail");
```

Comment:
I don't understand why this is the case. What is guaranteeing that the diff is always 1?

Reply:
(If the answer is: it's a side-effect of #5028 then that's good enough. I don't know how sinks work and that's fine.)

Reply:
Yes, it is guaranteed upstream, when the differential-dataflow diff is moved into the data.